Lib/subprocess.py

```diff
@@ -1589,6 +1611,8 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                             errpipe_read, errpipe_write,
                             restore_signals, start_new_session, preexec_fn)
                     self._child_created = True
```

Comment:
Best not to ever allow this to be set to True; if an asynchronous event happened and raised an exception (KeyboardInterrupt) here, the cleanup code would see this as True even though you meant it to be False.

The logic should probably look like:

```python
if _vxworks:
  if self.pid != -1:
    self._child_created = True
else:
  self._child_created = True
```

Reply:
All this said... why is `_vxwapi.rtp_spawn` not raising an exception rather than returning -1?  An exception is expected when a child process could not be created.  Not a silent return with pid == -1.

Reply:
more exception information raises in _vxwapi.rtp_spawn, since VxWorks has not used fork/execv, so only return -1